awx/main/db/profiled_pg/base.py

```diff
@@ -0,0 +1,101 @@
+import os
+import traceback
+import uuid
+
+from django.core.cache import cache
+from django.core.cache.backends.locmem import LocMemCache
+from django.db.backends.postgresql.base import DatabaseWrapper
+
+from awx.main.utils import memoize
+
+__loc__ = LocMemCache(str(uuid.uuid4()), {})
+__all__ = ['DatabaseWrapper']
+
+
+class RecordedQueryLog(object):
+
+    dest = '/var/lib/awx/profile'
+
+    def __init__(self, log, db):
+        self.log = log
+        self.db = db
+        self.threshold = cache.get('awx-profile-sql-threshold')
+
+    def append(self, query):
+        ret = self.log.append(query)
+        self.write(query)
+        return ret
+
+    def write(self, query):
+        if self.threshold is None:
+            return
+        seconds = float(query['time'])
+        if seconds >= self.threshold:
+            sql = query['sql']
+            if sql.startswith('EXPLAIN'):
+                return
+            if not os.path.isdir(self.dest):
+                os.makedirs(self.dest)
+            filename = '%.3fs-%s.sql' % (seconds, uuid.uuid4())
```

Comment:
In logging all queries to a file, I found that (1) logging to a per-process file was best and (2) I longed for the process name to be include the name of the process.

The pattern you have here follows the request/response logger pattern. In that context, we _know_ it's the uwsgi process so this naming scheme is good for that purpose.

Reply:
Should we maybe include the *pid* and *name* of the process in the filename?

Reply:
Yes. Also note if you open in some init code, the pid can change as a result of a fork() dup of the fd.

Reply:
@chrismeyersfsu @matburt what if we made a per-pid file instead of a per-query file, and a `---` between each blob we output so we you can visually separate them?

Reply:
yes

Reply:
So one tricky part about a file per pid is that if you've got a multi-threaded process appending to the same file, _now_ you've got to deal with file locking, and that _is_ going to be slow.

Reply:
https://docs.python.org/3/library/threading.html#threading.get_ident `threading.get_ident()`

Reply:
What if instead we wrote a per-pid .sqlite database that we could query?  Imagine having a per-process interface where you could do things like:

```sql
SELECT sql, time FROM queries WHERE time > 2 AND sql LIKE '%main_host%';
```